qutebrowser/keyinput/modeparsers.py

```diff
@@ -169,11 +172,56 @@ def __init__(self, win_id, mode, parent=None):
         """
         super().__init__(win_id, parent)
         self._read_config(mode)
+        self._orig_sequence = keyutils.KeySequence()
         self._mode = mode
+        self._ignore_next_key = False
 
     def __repr__(self):
         return utils.get_repr(self, mode=self._mode)
 
+    def handle(self, e, *, dry_run=False):
+        """Override to pass the chain through on NoMatch
+
+        Args:
+            e: the KeyPressEvent from Qt.
+            dry_run: Don't actually execute anything, only check whether there
+                     would be a match.
+
+        Return:
+            A self.Match member.
+        """
+        if keyutils.is_modifier_key(e.key()) or self._ignore_next_key:
+            self._ignore_next_key = self._ignore_next_key and dry_run
```

Comment:
I don't get this - why do you only want to set `self._ignore_next_key` if `dry_run` is set?